components/TicketsSection.js

```diff
@@ -67,23 +69,29 @@ function TicketsRow ({ title, date, isFull }) {
   )
 }
 
-function Button ({ children, href }) {
+function Button ({ children, href, date }) {
   return (
-    <a className='button' href={href}>
+    <a className={`button ${checkTicket(date) ? 'buynow' : 'waitbuy'}`} href={href}>
```

Comment:
Accessing the system date inside a component’s render function makes it impure. Instead of doing that, you can create a container that subscribes to the system time and passes the computed result through props.